domains/pallets/domain-registry/src/lib.rs

```diff
@@ -133,6 +133,11 @@ mod pallet {
     pub(super) type Domains<T: Config> =
         StorageMap<_, Twox64Concat, DomainId, DomainConfig<T>, OptionQuery>;
 
+    /// At which block the domain was created.
+    #[pallet::storage]
+    pub(super) type CreatedAt<T: Config> =
+        StorageMap<_, Twox64Concat, DomainId, T::BlockNumber, ValueQuery>;
```

Comment:
Make sense to use `OptionQuery`. Using a separate storage item could be better performance-wise so that we don't have to decode the entire DomainConfig when we only want to read `CreatedAt`, which is relatively frequent as we'll read it for each bundle.